src/server/job_func.c

```diff
@@ -423,12 +423,14 @@ job_alloc(void)
 
 #ifndef PBS_MOM
 	/* start accruing time from the time job was created */
-	pj->ji_wattr[(int)JOB_ATR_sample_starttime].at_val.at_long = (long) time_now;
-	pj->ji_wattr[(int)JOB_ATR_eligible_time].at_val.at_long = 0;
+	pj->ji_wattr[JOB_ATR_sample_starttime].at_val.at_long = (long) time_now;
+	pj->ji_wattr[JOB_ATR_eligible_time].at_val.at_long = 0;
+	pj->ji_wattr[JOB_ATR_eligible_time].at_flags |= ATR_VFLAG_SET;
+	pj->ji_wattr[JOB_ATR_sample_starttime].at_flags |= ATR_VFLAG_SET;
```

Comment:
Keep it along with its assignment (before JOB_ATR_eligible_time) for readability.